### PR TITLE
Docs: added less confusing explanation for func-style (fixes #12900)

### LINKED ISSUE #12900: Incorrect documentation for func-style
**Tell us about your environment**

* **ESLint Version:**
* **Node Version:**
* **npm Version:**

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js

```

```bash

```

**What did you expect to happen?**

The correct code example for the configuration `/*eslint func-style: ["error", "expression"]*/` shouldn't have an arrow function, as they're disallowed by default.

https://eslint.org/docs/rules/func-style#expression

**What actually happened? Please include the actual, raw output from ESLint.**

```
/*eslint func-style: ["error", "expression"]*/

var foo = function() {
    // ...
};

var foo = () => {};
```

**Are you willing to submit a pull request to fix this bug?**

Maybe.

### LINKED ISSUE #12924: arrow-body-style no braces for one-liner
**What rule do you want to change?**

`arrow-body-style`

**Does this change cause the rule to produce more or fewer warnings?**

Mmm, the same? I guess it depends on the format of your existing code.

**How will the change be implemented? (New option, new default behavior, etc.)?**

Probably a new option, something that gets across the idea of "always-multi-line-never-single-line".

**Please provide some example code that this change will affect:**

```js
// good
const valid_fields = fields.filter(field => field.is_valid);
const addresses = users.map((user) => {
  return user.address;
});

// bad
const valid_fields = fields.filter((field) => { return field.is_valid; });
```

**What does the rule currently do for this code?**

Depending on the configuration, it will require braces on one-liners or require that multi-line blocks don't have braces.

**What will the rule do after it's changed?**

Require that one-line arrow functions *don't* have braces and multi-line arrow functions *do* have braces.

**Are you willing to submit a pull request to implement this change?**

Willing, though it might be a while...

### PR DESCRIPTION
**Pull request by anikethsaha**
#### Prerequisites checklist

- [X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/master/CONTRIBUTING.md).
- [X] The team has reached consensus on the changes proposed in this pull request. If not, I understand that the evaluation process will begin with this pull request and won't be merged until the team has reached consensus.

#### What is the purpose of this pull request? (put an "X" next to an item)

- [X] Documentation update
- [ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
- [ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
- [ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
- [ ] Add autofixing to a rule
- [ ] Add a CLI option
- [ ] Add something to the core
- [ ] Other, please explain:

closes #12900

#### What changes did you make? (Give an overview)
Two commits

- added lock files to gitignore
- made docs changes for `func-style` rule

#### Is there anything you'd like reviewers to focus on?
#12900

### CODE REVIEW

**Comment by anikethsaha** [comment]
I am not sure whats the error in PR title !
Any help ? is it because its long ?

**Comment by mdjermanovic** [comment]
> I am not sure whats the error in PR title !
> Any help ? is it because its long ?

Thanks for the PR! Yes, the title is too long, max length is 72.

You can find the details about the message format here:

https://eslint.org/docs/developer-guide/contributing/pull-requests#step-2-make-your-changes

Also, can you please limit the scope of this PR to just fixing the documentation issue #12900, and maybe open another issue/PR for the `.gitignore` problem?

It's easier to manage and discuss unrelated changes in separate PRs.

**Comment by anikethsaha** [comment]
> Thanks for the PR! Yes, the title is too long, max length is 72.

I doubted that ! 💯 

> Also, can you please limit the scope of this PR to just fixing the documentation issue #12900, and maybe open another issue/PR for the .gitignore problem?

Sure !

**Comment by mdjermanovic** [comment]
Thanks! I also changed the PR title.

File: docs/rules/func-style.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ This rule has a string option:
 
 This rule has an object option for an exception:
 
-* `"allowArrowFunctions": true` (default `false`) allows the use of arrow functions (honoured only when using `declaration`)
+* `"allowArrowFunctions": true` (default `false`) allows the use of arrow functions (works only when using `declaration` in `func-style`'s options)
```
Comment (mdjermanovic) [inline docs/rules/func-style.md]: Maybe we should already mention here that the arrow functions are allowed for `"expression"`?

Something like: `"allowArrowFunctions": true` (default `false`) allows the use of arrow functions. This option applies only when the string option is set to `"declaration"` (arrow functions are always allowed when the string option is set to `"expression"`, regardless of this option).

Comment (anikethsaha) [reply docs/rules/func-style.md]: >  This option applies only when the string option is set to "declaration" (arrow functions are always allowed when the string option is set to "expression", regardless of this option).

Sure ! yes it makes much more sense

**Pull request merged**